R/pmcmc.R

```diff
@@ -0,0 +1,1029 @@
+  # Section 3 of pMCMC Wrapper: Sample PMCMC Results
+  #...........................................................
+  pmcmc_samples <- sample_pmcmc(pmcmc_results = pmcmc,
+                                burnin = burnin,
+                                n_chains = n_chains,
+                                n_trajectories = replicates,
+                                n_particles = n_particles,
+                                forecast_days = forecast)
+
+  #............................................................
+  # Section 4 of pMCMC Wrapper: Tidy Output
+  #...........................................................
+  #..................
+  # Pull Sampled results and "recreate" squire models
+  #..................
+  if (inherits(squire_model, "stochastic")) {
+
+    # create a fake run object and fill in the required elements
+    r <- squire_model$run_func(country = country,
+                               contact_matrix_set = contact_matrix_set,
+                               tt_contact_matrix = tt_contact_matrix,
+                               hosp_bed_capacity = hosp_bed_capacity,
+                               tt_hosp_beds = tt_hosp_beds,
+                               ICU_bed_capacity = ICU_bed_capacity,
+                               tt_ICU_beds = tt_ICU_beds,
+                               population = population,
+                               replicates = 1,
+                               time_period = max(tt_contact_matrix,tt_hosp_beds,tt_ICU_beds,1))
+
+    # first let's add our pmcmc results for a nice return
+    # we'll save our inputs so it is easy to recreate later
+    r$pmcmc_sample_inputs <- pmcmc_samples$inputs
+    # and add the parameters that changed between each simulation, i.e. posterior draws
+    r$trajectory_parameters <- pmcmc_samples$sampled_PMCMC_Results
+    # as well as adding the pmcmc chains so it's easy to draw from the chains again in the future
+    r$pmcmc_results <- pmcmc
+
+
+    # then let's create the output that we are going to use
+    names(pmcmc_samples)[names(pmcmc_samples) == "trajectories"] <- "output"
+    dimnames(pmcmc_samples$output) <- list(dimnames(pmcmc_samples$output)[[1]], dimnames(r$output)[[2]], NULL)
+    r$output <- pmcmc_samples$output
+
+    # and adjust the time as before
+    full_row <- match(0, apply(r$output[,"time",],2,function(x) { sum(is.na(x)) }))
+    saved_full <- r$output[,"time",full_row]
+    for(i in seq_len(replicates)) {
+      na_pos <- which(is.na(r$output[,"time",i]))
+      full_to_place <- saved_full - which(rownames(r$output) == as.Date(max(data$date))) + 1L
+      if(length(na_pos) > 0) {
+        full_to_place[na_pos] <- NA
+      }
+      r$output[,"time",i] <- full_to_place
+    }
+
+  } else if (inherits(squire_model, "deterministic")) {
+
+
+    r <- list("output" = pmcmc_samples$trajectories)
+    # same as above, add in pmcmc items
+    r$pmcmc_sample_inputs <- pmcmc_samples$inputs
```

Comment:
same as above just make these `inputs and `replicate_parameters`

Reply:
done!